core/src/main/java/io/substrait/relation/ProtoRelConverter.java

```diff
@@ -527,6 +532,29 @@ private Rel newHashJoin(HashJoinRel rel) {
     return builder.build();
   }
 
+  private NestedLoopJoin newNestedLoopJoin(NestedLoopJoinRel rel) {
+    Rel left = from(rel.getLeft());
+    Rel right = from(rel.getRight());
+    Type.Struct leftStruct = left.getRecordType();
+    Type.Struct rightStruct = right.getRecordType();
+    Type.Struct unionedStruct = Type.Struct.builder().from(leftStruct).from(rightStruct).build();
+    var converter = new ProtoExpressionConverter(lookup, extensions, unionedStruct, this);
+    var builder =
+        NestedLoopJoin.builder()
+            .left(left)
+            .right(right)
+            .condition(converter.from(rel.getExpression()))
```

Comment:
I think we still want your original code that checks if the condition is null and then sets the condition to True if it is. The `expression` field in the protobuf isn't marked as required, so we can't assume it's present when reading the protos in.

Reply:
How I'm thinking about this is that this is _effectively_ required, because when it is null it should be treated as True. But we can handle that behaviour at our serialisation boundary and have it be properly required internally, which makes it easier to work with within our code.

Reply:
Thanks for the explanation! +1